switools/swisignature.py

```diff
@@ -144,9 +160,123 @@ def signSwiHandler( args ):
+      sys.exit( SWI_SIGN_RESULT.ERROR_NOT_A_SWI )
+
+   try:
+      optims = signaturelib.getOptimizations( swi, workDir )
+      if optims is None or len( optims ) == 1 or "DEFAULT" in optims:
+         # legacy case of a single rootfs image
+         # maybe need to use a remote signing service (new feature in v1.2)
+         if not signatureFile and not signingKeyFile:
+            signatureFile = "%s/sig" % workDir
+            getSignatureFile( swi, signatureFile ) # never returns in case of fail
+         if not signatureFile and not signingKeyFile:
+            print ( 'Error: without signing key we need a signature file: '
+                    'run "swi-signature prepare" first and have the digest it prints '
+                    'signed and passed to this command.' )
+            sys.exit( SWI_SIGN_RESULT.ERROR_NO_SIGNATURE_FILE_PROVIDED )
+         if not signatureFile and signingKeyFile:
+            sha256 = prepareSwi( swi=swi, outfile=None, forceSign=True )
+            print( "%s sha256: %s" % ( os.path.basename( swi ), sha256 ) )
+         # insert the signatureFile into the swi, or use the signing key to create the signatureFile first
+         return signSwi( swi, signingCertFile, rootCaFile,
+                         signatureFile=signatureFile, signingKeyFile=signingKeyFile )
+
+      print( "Optimizations in %s: %s" % ( swi, " ".join( optims ) ) )
+      signaturelib.extractSwadapt( swi, workDir )
+      optimSigFiles = []
+      for optim in optims:
+         optimImage = "%s/%s.swi" % ( workDir, optim )
+         # Adapt swi (extract an optimized image)
+         os.system("%s/swadapt %s %s %s" % ( workDir, swi, optimImage, optim ) )
+         if not signingKeyFile: # need to use a remote signing service
+            signatureFile = "%s/sig" % workDir
+            getSignatureFile( optimImage, signatureFile ) # never returns in case of fail
+         else:
+            sha256 = prepareSwi( swi=optimImage, outfile=None, forceSign=True )
+            print( "%s sha256: %s" % ( optim, sha256 ) )
+         # Sign optimized swi using provided key or provided full fledged signature file
+         signSwi( optimImage, signingCertFile, rootCaFile,
+                  signatureFile=signatureFile, signingKeyFile=signingKeyFile )
+
+         # Extract the signature from the optim and call it <optim>.sig
+         optimSigFile = "%s.signature" % optim
+         optimSigPath = "%s/%s" % ( workDir, optimSigFile )
+         optimSigFiles.append( os.path.basename( optimSigPath ) )
+         extractSignature( optimImage, optimSigPath )
+
+         os.remove( optimImage )
+
+      # update the source swi with the signatures of its "baby" swis (optims)
+      print( "Adding signature files to %s: %s" % ( swi, " ".join( optimSigFiles ) ) )
+      ret = os.system( "src=$(readlink -f %s); cd %s; "
+                       "zip -q -0 -X $src %s" % ( swi, workDir,
+                       " ".join( optimSigFiles ) ) )
+      if ret != 0:
```

Comment:
```suggestion
      try:
          subprocess.check_call( [ 'zip', '-q', '-0', '-X', swi, optimSigFiles
                                  ], cwd=workDir )
      except subprocess.CalledProcessError:
```

Reply:
Another mistake: `optimSigFIles` is a list so this needs to be `['zip', ..., swi] + optimSigFiles`